pkg/features/features.go

```diff
@@ -38,22 +43,45 @@ const (
 	GPTBasedUUID Feature = "GPTBasedUUID"
 )
 
-// DefaultFeatureGates is the list of feature gates that are added by default
-var DefaultFeatureGates = []Feature{
+// supportedFeatures is the list of supported features. This is used while parsing the
+// feature flag
+var supportedFeatures = []Feature{
 	GPTBasedUUID,
 }
 
+// defaultFeatureGates is the default features that will be applied to the application
+var defaultFeatureGates = map[Feature]bool{
+	GPTBasedUUID: false,
+}
+
+// featureGate is a type that implements the FeatureGate interface
+type FeatureGate map[Feature]bool
+
+// DefaultFeatureGate is the global feature gate that can be used to check if a feature is enabled
+// or disabled
+var DefaultFeatureGates = NewFeatureGate()
```

Comment:
n_: DefaultFeatureGates => FeatureGates or a generic name as it contains both default and those enabled via ENV etc,. right?

Reply:
`FeatureGates` was not used because there is already a `FeatureGate` type in the same package.

Do you have suggestion for any other generic names?

Reply:
`FeatureGate` will renamed to `featureFlag`, and `DefaultFeatureGates` to `FeatureGates`